src/frequenz/client/dispatch/types.py

```diff
@@ -21,16 +32,180 @@
+
+    AC = PBEvChargerType.EV_CHARGER_TYPE_AC
+    """AC EV charger."""
+
+    DC = PBEvChargerType.EV_CHARGER_TYPE_DC
+    """DC EV charger."""
+
+    HYBRID = PBEvChargerType.EV_CHARGER_TYPE_HYBRID
+    """Hybrid EV charger."""
+
+
+class BatteryType(IntEnum):
+    """Enum representing the type of battery."""
+
+    UNSPECIFIED = PBBatteryType.BATTERY_TYPE_UNSPECIFIED
+    """Unspecified type of battery."""
+
+    LI_ION = PBBatteryType.BATTERY_TYPE_LI_ION
+    """Lithium-ion battery."""
+
+    NA_ION = PBBatteryType.BATTERY_TYPE_NA_ION
+    """Sodium-ion battery."""
+
+
+class InverterType(IntEnum):
+    """Enum representing the type of inverter."""
+
+    UNSPECIFIED = PBInverterType.INVERTER_TYPE_UNSPECIFIED
+    """Unspecified type of inverter."""
+
+    BATTERY = PBInverterType.INVERTER_TYPE_BATTERY
+    """Battery inverter."""
+
+    SOLAR = PBInverterType.INVERTER_TYPE_SOLAR
+    """Solar inverter."""
+
+    HYBRID = PBInverterType.INVERTER_TYPE_HYBRID
+    """Hybrid inverter."""
+
+
+@dataclass(frozen=True)
+class TargetCategory:
+    """Represents a category and optionally a type."""
+
+    target: ComponentCategory | BatteryType | EvChargerType | InverterType
+    """The target category of the dispatch.
+
+    Implicitly derived from the types.
+    """
+
+    @property
+    def category(self) -> ComponentCategory:
+        """Get the category of the target.
+
+        Returns:
+            The category of the target.
+        """
+        match self.target:
+            case ComponentCategory():
+                return self.target
+            case BatteryType():
+                return ComponentCategory.BATTERY
+            case EvChargerType():
+                return ComponentCategory.EV_CHARGER
+            case InverterType():
+                return ComponentCategory.INVERTER
+
+    @property
+    def type(self) -> BatteryType | EvChargerType | InverterType | None:
+        """Get the type of the category.
+
+        Returns:
+            The type of the category.
+        """
+        match self.target:
+            case BatteryType() | EvChargerType() | InverterType():
+                return self.target
+            case _:
+                return None
+
+
+class TargetIds(frozenset[int]):
+    """A set of target component IDs.
+
+    This is a frozen set, so it is immutable.
+    """
+
+    def __new__(cls, *ids: SupportsInt) -> "TargetIds":
+        """Create a new TargetIds instance.
+
+        Args:
+            *ids: The target IDs to initialize.
+
+        Returns:
+            A new TargetIds instance.
+        """
+        # Convert all provided ids to integers before creating the frozenset
+        processed_ids = tuple(int(id_val) for id_val in ids)
+        return super().__new__(cls, processed_ids)
+
+
+# Define the union of types that can be passed to TargetCategories constructor
+TargetCategoryInputType = (
+    TargetCategory | ComponentCategory | BatteryType | InverterType | EvChargerType
+)
```

Comment:
Usually it is a good idea to document type aliases too so they appear in the docs.
```suggestion
)
"""The union of all types that can be used as target categories."""
```